### DISCUSSION TITLE
can't get barebones example building

### DISCUSSION THREAD

**Discussion by goosetherumfoodle** [90-008/nix-cargo-integration#109] [Q&A] [answered]
I'm trying to just get this working with rust-hello and running into an `flake does not provide attribute 'packages.x86_64-linux.default'` error.

steps:
`cargo init rust-hello`
`cd rust-hello`
`nix flake init -t github:yusdacra/nix-cargo-integration`
`echo "[package.metadata.nix]" >> Cargo.toml`
`echo "build = true" >> Cargo.toml`
`git add .`
`nix build`

Error:
```
error: flake 'git+file:///<path>/rust-hello' does not provide attribute 'packages.x86_64-linux.default' or 'defaultPackage.x86_64-linux'
```

Adding `systems = ["x86_64-linux"]` to flake.nix doesn't change the error.

**Comment by 90-008** [comment, marked as answer]
`nix-cargo-integration` doesn't provide any default packages. You need to use the `outputs.defaults.package` option to set a default package. Like so:

```nix
# flake.nix
{
  # ...
  outputs = inputs:
    inputs.nci.lib.makeOutputs {
      # ...
      config = common: {
        outputs.defaults = {
          package = "rust-hello";
        };
      };
      # ...
    };
}
```

**Reply by goosetherumfoodle** [reply]
Ah ok thanks. 

After that step I'm running into the issue of not having a `Cargo.lock`. I see that nci is using dream2nix under the hood, and I'm getting the message (that I think is from dream2nix) about having an unresolved impurity. But `nix flake show` is only giving me the "rust-hello" package, not the "resolveImpure" package that dream2nix exposes. Do I have to do something additional to be able to run `resolveImpure`?

**Reply by 90-008** [reply]
You just need to create a Cargo.lock, so just run cargo generate-lockfile for your project.
